Update README rules and related content for amqp-common package

ramya0820 opened the pull request:
This PR is to verify current doc rules are setup fine.
More context available at #1583 

This closes #2735

scbedd commented:
Heyo Ramya, saving you some frustration on debugging these:

First error:
`Azure client library for X` is failing due to expecting "for JS" instead of "for UAMQP operations"
I updated the expected regex line to this: `- ^Azure .+ client library for (JS|AMQP operations)` to get it to pass. I can either push a commit to your PR or leave off, up to you!

Second error:
`Key Concepts` is expected to be a `#` or `##`. AKA H1 or H2 header. That's why it's not detecting it.

ramya0820 commented:
> Heyo Ramya, saving you some frustration on debugging these:
> 
> First error:
> `Azure client library for X` is failing due to expecting "for JS" instead of "for UAMQP operations"
> I updated the expected regex line to this: `- ^Azure .+ client library for (JS|AMQP operations)` to get it to pass. I can either push a commit to your PR or leave off, up to you!
> 
> Second error:
> `Key Concepts` is expected to be a `#` or `##`. AKA H1 or H2 header. That's why it's not detecting it.

Thank you for the context! 
I've updated the rule and amqp-common README for this PR, please take a look when you can.
